### PR TITLE
[BEAM-4617] Adds a dependencies guide to Beam Website

### PR DESCRIPTION
**Pull request by chamikaramj**
This is based on following discussions in dev list and the doc.

https://lists.apache.org/thread.html/8738c13ad7e576bc2fef158d2cc6f809e1c238ab8d5164c78484bf54@%3Cdev.beam.apache.org%3E

https://lists.apache.org/thread.html/8b9b3768adfc40d3527d1ce5e8a51d90e5782a348a3abfb9e5dc85ef@%3Cdev.beam.apache.org%3E

https://docs.google.com/document/d/15m1MziZ5TNd9rh_XN0YYBJfYkt0Oj-Ou9g0KFDPL2aA/edit

### CODE REVIEW

File: src/contribute/dependencies.md [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+
+__Dependency declarations may identify owners that are responsible for upgrading respective dependencies.__
+
+Owners can be mentioned in a comment. Blocking JIRAs will be initially assigned to these owners (if available). Release manager may choose to re-assign these JIRAs. A dependency may have more than one declared owner and in this case the JIRA will be assigned to one of the owners mentioned.
+
+__Dependencies of Java SDK components that may cause issues to other components if leaked should be vendored.__
+
+[Vendoring](https://www.ardanlabs.com/blog/2013/10/manage-dependencies-with-godep.html) is the process of creating copies of third party dependencies. Combined with repackaging, veodoring will allow Beam components to depend on third party libraries without causing conflicts to other components. Vendoring should be done in a case-by-case basis since this can increase the total number of dependencies deployed in user's enviroment.
```
Comment (aaltay) [inline src/contribute/dependencies.md]: veodoring -> vendoring

**Comment by chamikaramj** [comment]
Retest this please

**Comment by chamikaramj** [comment]
Retest this please

**Comment by chamikaramj** [comment]
@asfgit merge

**Pull request closed without merging**